### PR TITLE
Adds redux-persist and removes redux-session

### PR DESCRIPTION
**Pull request by mcousillas6**
Adds redux persist basic configuration for local storage.

### Changes
- Added basic whitelist configuration for the reducers that we want to persist.
- Removed react-native-redux-session since it was redundant to store things twice.
- Migrated all userActions that were using `react-native-redux-session`
- Updates api middleware to store session.

### Advantages
- Easier control for session dispatchs. With `react-native-redux-session` actions were dispatched unexpectedly in some cases.
- Ability to have cache for other reducers/entities (a feed for example).
- Migration control over persisted data.

### CODE REVIEW

File: src/actions/userActions.js [hunk 1]
```diff
@@ -54,43 +56,37 @@ const signUpError = error => ({
...
     const { data } = await userService.login({ user });
-    await sessionService.saveUser(data.user);
-    dispatch(loginSuccess());
-  } catch (err) {
-    dispatch(loginError(err));
-    throw new SubmissionError({
-      _error: err.data.error,
-    });
+    dispatch(loginSuccess(data.user));
+  } catch (error) {
```
Comment (mariosouto) [inline src/actions/userActions.js]: you could destructure `data` here

File: src/actions/userActions.js [hunk 2]
```diff
@@ -54,43 +56,37 @@ const signUpError = error => ({
...
     dispatch(logoutRequest());
     await userService.logout();
-    sessionService.deleteSession();
-    sessionService.deleteUser();
     dispatch(logoutSuccess());
-  } catch (err) {
-    dispatch(logoutError(err));
-    throw err.data.error;
+  } catch (error) {
+    dispatch(logoutError(error.data.error));
```
Comment (mariosouto) [inline src/actions/userActions.js]: Same

File: src/actions/userActions.js [hunk 3]
```diff
@@ -54,43 +56,37 @@ const signUpError = error => ({
...
-    sessionService.saveUser(data.user);
-    dispatch(signUpSuccess());
-  } catch (err) {
-    dispatch(signUpError(err));
-    throw new SubmissionError({
-      _error: err.data.error,
-    });
+    dispatch(signUpSuccess(data.user));
+  } catch (error) {
+    dispatch(signUpError(error.data.error));
```
Comment (mariosouto) [inline src/actions/userActions.js]: Same

File: src/store/configureStore.js [hunk 4]
```diff
@@ -1,29 +1,39 @@
...
-      predicate: (getState, { type }) => !_.startsWith(type, '@@redux-form'),
     });
     middlewares.push(logger);
   }
 
+  const persistedReducer = persistReducer(persistConfig, AppReducer);
+
   const store = createStore(
-    AppReducer,
+    persistedReducer,
```
Comment (peter7z) [inline src/store/configureStore.js]: Do we want to persist every piece of the app state by default?

File: src/services/userService.js [hunk 5]
```diff
@@ -1,17 +1,17 @@
...
   }
 
-  static signUp(user) {
+  signUp(user) {
     return api.post('/users', user);
   }
 }
 
-export default UserService;
+export default new UserService();
```
Comment (peter7z) [inline src/services/userService.js]: Why this change? This class holds no state.

File: src/screens/MainScreen/index.js [hunk 6]
```diff
@@ -21,7 +21,7 @@ const MainScreen = ({ navigation }) => {
 
   return (
     <View style={styles.container}>
-      <Text>Hey {email}, you&#39;re logged in!</Text>
+      <Text>Hey {email || ''}, you&#39;re logged in!</Text>
```
Comment (peter7z) [inline src/screens/MainScreen/index.js]:
```suggestion
      <Text>Hey{` ${email}` || ''}, you&#39;re logged in!</Text>
```

File: src/store/configureStore.js (hunk 4 above)
Comment (mcousillas6) [reply src/store/configureStore.js]: Nope, we have a whitelist in `cofingureStore.js` that specifies which reducers are going to be persisted, by default I just left `session`

File: src/services/userService.js (hunk 5 above)
Comment (mcousillas6) [reply src/services/userService.js]: For consistency sake. The `ApiService` exports a singleton, so it made more sense to keep all services exposing the same interface.

File: src/reducers/sessionReducer.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  user: null,
+  info: null,
+};
+
+const handleLoginSuccess = (state, { user }) => {
+  state.user = user;
+};
+
+const handleLogoutSuccess = state => {
+  state.user = null;
```
Comment (aguscha333) [inline src/reducers/sessionReducer.js]: I think here you can just do `return initialState;`

Comment (mcousillas6) [reply src/reducers/sessionReducer.js]: I think I can't since this is an immer callback and we need to mutate the state without returning it.

Comment (aguscha333) [reply src/reducers/sessionReducer.js]: I am using immer and have this on logout success:
image

Comment (mcousillas6) [reply src/reducers/sessionReducer.js]: neat, will change it then
